### PR TITLE
[TT-4852] Add disabled flag to body transform endpoint plugin

### PR DESCRIPTION
**Pull request by furkansenharputlu**
This PR adds `disabled` flag to transform request/response body endpoint plugins so that it is possible to disable a configured plugins.

### CODE REVIEW

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/3831/merge
Commit: db138d1faa5e62642775c50d3d6c350d63c3db23 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/1993470176)

**Review by jeffy-mathew — APPROVED** [review]

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/3831/merge
Commit: 0163415ebea946407c36e283ac86de0e607cee71 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2773308908)

File: gateway/mw_transform_test.go [hunk 1]
```diff
@@ -349,3 +350,41 @@ func TestBodyTransformCaseSensitivity(t *testing.T) {
 		assert("/Get", "/Get", `{"http_method":"GET"}`)
 	})
 }
+
+func TestTransformRequestBody(t *testing.T) {
```
Comment (titpetric) [inline gateway/mw_transform_test.go:354]: What if instead you unit test the APIDefinitionLoader `compileTransformPathSpec`; we need more unit tests and less integration tests, and this is a case that is unit testable...

**Review by titpetric — APPROVED** [review]

**Review by urbanishimwe — APPROVED** [review]

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/heads/master
Commit: f6ac1cb91b2b0db84af0120b1a281346280d93f1 Add disabled flag to body transform endpoint plugin (#3831)
Triggered by: push (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2773464404)

**Pull request merged**